eth/downloader/downloader.go

```diff
@@ -219,6 +222,8 @@ func New(checkpoint uint64, stateDb ethdb.Database, mux *event.TypeMux, chain Bl
 		SnapSyncer:     snap.NewSyncer(stateDb),
 		stateSyncStart: make(chan *stateSync),
 	}
+	dl.skeleton = newSkeleton(stateDb, dl.peers, dropPeer, newBeaconBackfiller(dl, success))
```

Comment:
This starts a goroutine in the skeleton, which will wait for you to call `Terminate` on the skeleton. AFAICT you never do? 
This _won't_ cause a hang, since you're also not adding to any `wg`, but shutdown will just tear it down. Is that intentional?

Reply:
Hmm, good catch. Added the skeleton terminate call into the downloader.Terminate